### PR TITLE
packaging: Build from source if the clh release binary is missing

### PR DESCRIPTION
**Pull request by likebreath**
This patch add fall-back code path that builds cloud-hypervisor static
binary from source, when the downloading of cloud-hypervisor binary is
failing. This is useful when we experience network issues, and also
useful for upgrading clh to non-released version.

Together with the changes in the tests repo (kata-containers/tests#2862),
the Jenkins config file is also updated with new Execute shell script
for the clh CI in the kata-containers repo. Those two changes fix the
regression on clh CI here. Please check details in the issue below.

Fixes: kata-containers/tests#2858

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

**Review by amshinde — APPROVED** [review]

**Comment by amshinde** [comment]
@jcvenegas @bergwolf PTAL

**Comment by likebreath** [comment]
The travis-CI failing is strange: `test mount::tests::test_init_rootfs ... FAILED`. It was failing on either one of the `target_branch=$TRAVIS_BRANCH` or `RUST_BACKTRACE=1` from time to time (w/ the same change that's not related).

/test-clh
/test-ubuntu
/test-fodera

**Comment by likebreath** [comment]
Both CLH and qemu CI are passing, but we are facing two issues that should not be relevent:
1) Travis-CI failure as mentioned above
2) the fedora CI never returns..

Any inputs? @kata-containers/kata-containers

File: tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh [hunk 1]
```diff
@@ -18,7 +18,27 @@ cloud_hypervisor_version="${cloud_hypervisor_version:-}"
...
-curl --fail -L ${cloud_hypervisor_binary} -o cloud-hypervisor
+    curl --fail -L ${cloud_hypervisor_binary} -o cloud-hypervisor || return 1
+}
+
+build_clh_from_source() {
+    info "Build ${cloud_hypervisor_repo} version: ${cloud_hypervisor_version}"
+    repo_dir=$(basename "${cloud_hypervisor_repo}")
+    repo_dir="${repo_dir//.git}"
+    [ -d "${repo_dir}" ] || git clone "${cloud_hypervisor_repo}"
+    cd "${repo_dir}"
```
Comment (devimc) [inline tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh:33]: I'd prefer to use `pushd` and `popd` before returning

File: tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh [hunk 2]
```diff
@@ -18,7 +18,27 @@ cloud_hypervisor_version="${cloud_hypervisor_version:-}"
...
+    git fetch || true
+    git checkout "${cloud_hypervisor_version}"
+    ./scripts/dev_cli.sh build --release --libc musl
```
Comment (devimc) [inline tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh:36]: I wonder if we don't need to install the build dependencies first .. :thinking:

**Review by devimc — APPROVED** [review]
thanks @likebreath - just two nits that can be ignored

File: tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh (hunk 1 above)
Comment (likebreath) [reply tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh]: Good point. Just updated the PR. Thanks.

File: tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh (hunk 2 above)
Comment (likebreath) [reply tools/packaging/static-build/cloud-hypervisor/build-static-clh.sh:36]: This is the script provided by cloud-hypervisor (clh) to build clh w/o installing any dependencies (by using its own custom container image). :)

**Comment by likebreath** [comment]
All required CI jobs are passing, and also for the CLH CI jobs. Merge the PR now.

**Pull request merged**